torchmultimodal/modules/encoders/albef_multimodal_encoder.py

```diff
@@ -87,30 +85,46 @@ def __init__(
     ) -> None:
         super().__init__()
         # the attention block computes the self attention on text embeddings
-        self.attention = ALBEFTransformerAttention(
-            hidden_size, num_attention_heads, layer_norm_eps
+        self.attention = MultiHeadAttention(
+            dim_q=hidden_size,
+            dim_kv=hidden_size,
+            n_head=num_attention_heads,
+            causal=False,
         )
         # the cross_attention block computes the cross attention on image and text embeddings
-        self.cross_attention = ALBEFTransformerAttention(
-            hidden_size, num_attention_heads, layer_norm_eps
+        self.cross_attention = MultiHeadAttention(
+            dim_q=hidden_size,
+            dim_kv=hidden_size,
+            n_head=num_attention_heads,
+            causal=False,
         )
+        self.attention_layer_norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+        self.cross_attention_layer_norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
         self.dense1 = nn.Linear(hidden_size, intermediate_size)
         self.transform_act_fn = transform_act_fn
         self.dense2 = nn.Linear(intermediate_size, hidden_size)
-        self.layer_norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
+        self.ffn_layer_norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
 
     def forward(
         self,
         hidden_states: Tensor,
         attention_mask: Tensor,
         encoder_hidden_states: Tensor,
     ) -> Tensor:
-        attention_output = self.attention(hidden_states, attention_mask)
-        attention_output = self.cross_attention(
-            attention_output, attention_mask, encoder_hidden_states
+        attention_output = self.attention(hidden_states, attention_mask=attention_mask)
+        attention_norm_output = self.attention_layer_norm(
+            attention_output + hidden_states
         )
-        dense1_output = self.dense1(attention_output)
+        cross_attention_output = self.cross_attention(
+            attention_norm_output,
+            kv=encoder_hidden_states,
+            attention_mask=attention_mask,
```

Comment:
remove line 121 (I actually set `attention_mask=None` in the original `ALBEFTransformerSelfAttention` class when computing cross attention)

Reply:
ok you're right this was the issue. I missed the fact that the original class ignored the attention mask so I need to explicitly make sure this is None